tlsutil/config.go

```diff
@@ -224,13 +226,7 @@ func (c *Configurator) commonTLSConfig(additionalVerifyIncomingFlag bool) (*tls.
 	}
 
 	tlsConfig := &tls.Config{
-		ClientAuth:         tls.NoClientCert,
-		InsecureSkipVerify: c.base.skipBuiltinVerify(),
-		ServerName:         c.base.ServerName,
-	}
-
-	if tlsConfig.ServerName == "" {
-		tlsConfig.ServerName = c.base.NodeName
+		InsecureSkipVerify: !c.base.VerifyServerHostname,
```

Comment:
Since `ServerName` is not used here anymore, this only depends on `VerifyServerHostname` now.